### DISCUSSION TITLE
Problem with overlapping patches when utilizing the "combine_tensor_patches" function

### DISCUSSION THREAD

**Discussion by kiratben** [kornia/kornia#2513] [Questions]
I encounter a problem with overlapping patches when utilizing the "combine_tensor_patches" function. I'm encountering a NotImplementedError: Only stride == window_size is supported, but I have (1, 1) stride and (3, 3) window size. If you have any suggestions for resolving this issue, please let me know.

**Comment by edgarriba** [comment]
can you how do you use the function ? the error you report seems to be raising this one
https://github.com/kornia/kornia/blob/master/kornia/contrib/extract_patches.py#L246

**Reply by kiratben** [reply]
`patches = kornia.contrib.extract_tensor_patches(x, window_size=(3, 3), stride=(1, 1))`
It seems that the issue arises when I use “combine_tensor_patches.”
`output = kornia.contrib.combine_tensor_patches(patches, original_size=x.shape[2], window_size=(3,3), stride=(1,1))`

**Reply by edgarriba** [reply]
what's your input `x` shape ?

**Reply by kiratben** [reply]
My input data is organized in a tensor with dimensions B (batch size), C (number of channels), H (height), and W (width), specifically in the shape of 32, 32, 64, 64.

**Reply by edgarriba** [reply]
not sure if the input shape should be divisible by window_size and stride? @shijianjian @ashnair1

**Reply by ashnair1** [reply]
Currently `CombineTensorPatches` only works if stride == window_size. In your case, your stride and window_size is (1,1) and (3,3) respectively. This is a known weakness of the implementation. #2650 aims to lift this restriction.
